net2/http2/load_balanced_pool.go

```diff
@@ -90,15 +90,13 @@ func (pool *LoadBalancedPool) Update(instanceInfos []LBPoolInstanceInfo) {
 	pool.lock.Lock()
 	defer pool.lock.Unlock()
 	newInstances := make(map[string]*instancePool)
-	newInstanceList := make(instancePoolSlice, len(instanceInfos))
-	for i, instanceInfo := range instanceInfos {
-		var instance *instancePool
-		var ok bool
-		if instance, ok = pool.instances[instanceInfo.Addr]; !ok {
-			instance = pool.newInstancePool(instanceInfo)
+	var newInstanceList instancePoolSlice
+	for _, instanceInfo := range instanceInfos {
+		if _, ok := newInstances[instanceInfo.Addr]; !ok {
+			instance := pool.newInstancePool(instanceInfo)
```

Comment:
zviad brought up a good point,  we only want to create a new pool when it's not already in pool.instances